go-fuzz/internal/writerset/writerset.go

```diff
@@ -0,0 +1,98 @@
+// Package writerset implements a mechanism to add and remove writers from a construct
+// similar to io.MultiWriter. Vendored from github.com/stephens2424/writerset
+package writerset
+
+import (
+	"io"
+	"net/http"
+	"sync"
+)
+
+// WriterSet wraps multiple writers like io.MultiWriter, but such that individual
+// writers are easy to add or remove as necessary.
+type WriterSet struct {
+	m  map[io.Writer]chan error
+	mu *sync.Mutex
+}
+
+// New initializes a new empty writer set.
+func New() *WriterSet {
+	return &WriterSet{
+		m:  make(map[io.Writer]chan error),
+		mu: &sync.Mutex{},
+	}
+}
+
+// Add ensures w is in the set.
+func (ws *WriterSet) Add(w io.Writer) <-chan error {
+	ws.mu.Lock()
+	defer ws.mu.Unlock()
+
+	c, ok := ws.m[w]
```

Comment:
On a lot of the ones about writerset, I'm going to have this same comment: I wrote that package to be a tad more general than the use cases here and it's hosted as an independent library at github.com/stephens2424/writerset, but I've vendored it here (manually). So yes, parts of this aren't used in this project, but I think it's fitting for them to stay around. Of all the possible approaches to this, I think some type of vendoring makes sense, but if you'd rather pare down this to just what's used here, we can do that, but I'd like to double check before going ahead.